### PR TITLE
AP_Mount/RC_Channel: viewpro supports setting active lens

### PR DESCRIPTION
**Pull request by rmackay9**
This allows users to change the active video feed (aka "lens") using an auxiliary switch.  This is only implemented for the ViewPro backend.

There are also some other unrelated improvements to the Viewpro driver:

1. recording state is taken from the gimbal instead of assuming the start recording is always successful.  This means the state will remain correct in case of SD card failure or when another interface is used to start/stop recording.
2. zoom level, model name and firmware version are retrieved from the gimbal and displayed to the user and reported to the GCS
3. packet structures are filled in all at once as requested by @peterbarker
4. minor capitalisation fix to some enum values

This has been tested on real hardware.

### CODE REVIEW

File: libraries/AP_Mount/AP_Mount_Viewpro.cpp [hunk 1]
```diff
@@ -574,25 +685,28 @@ bool AP_Mount_Viewpro::send_m_ahrs()
...
+            pitch_be: htobe16(-degrees(AP::ahrs().get_pitch()) * AP_MOUNT_VIEWPRO_DEG_TO_OUTPUT),   // vehicle pitch angle.  1bit=360deg/65536
+            roll_be: htobe16(degrees(AP::ahrs().get_roll()) * AP_MOUNT_VIEWPRO_DEG_TO_OUTPUT),      // vehicle roll angle.  1bit=360deg/65536
+            yaw_be: htobe16(degrees(AP::ahrs().get_yaw()) * AP_MOUNT_VIEWPRO_DEG_TO_OUTPUT),        // vehicle yaw angle.  1bit=360deg/65536
+            date_be: htobe16(date),                 // bit0~6:year, bit7~10:month, bit11~15:day
+            seconds_utc: {uint8_t((second_hundredths & 0xFF0000ULL) >> 16), // seconds * 100 MSB.  1bit = 0.01sec
+                          uint8_t((second_hundredths & 0xFF00ULL) >> 8),    // seconds * 100 next MSB.  1bit = 0.01sec
+                          uint8_t(second_hundredths & 0xFFULL)},            // seconds * 100 LSB.  1bit = 0.01sec
+            gps_yaw_be: htobe16(vel_yaw_deg * AP_MOUNT_VIEWPRO_DEG_TO_OUTPUT),  // GPS yaw
+            position_mark_bitmask: 0x0F,            // bit0:new position, bit1:clock fix calced, bit2:horiz calced, bit3:alt calced
+            latitude_be: htobe32(loc.lat),          // latitude.  1bit = 10e-7
```
Comment (tridge) [inline libraries/AP_Mount/AP_Mount_Viewpro.cpp:701]: check HUD with -ve lat and -ve alt

File: libraries/AP_Mount/AP_Mount_Viewpro.cpp [hunk 2]
```diff
@@ -574,25 +685,28 @@ bool AP_Mount_Viewpro::send_m_ahrs()
...
+            longitude_be: htobe32(loc.lng),         // longitude.  1bit = 10e-7
+            height_be: htobe32(alt_amsl_cm * 10),   // height.  1bit = 1mm
```
Comment (tridge) [inline libraries/AP_Mount/AP_Mount_Viewpro.cpp:703]: check -ve height

**Review by tridge — DISMISSED** [review]

File: libraries/AP_Mount/AP_Mount_Viewpro.cpp [hunk 3]
```diff
@@ -413,9 +471,12 @@ bool AP_Mount_Viewpro::set_lock(bool lock)
     }
 
     // fill in packet
-    A1Packet a1_packet {};
-    a1_packet.content.frame_id = FrameId::A1;
-    a1_packet.content.servo_status = lock ? ServoStatus::follow_yaw_disable : ServoStatus::follow_yaw;
+    A1Packet a1_packet {
```
Comment (rmackay9) [inline libraries/AP_Mount/AP_Mount_Viewpro.cpp:474]: @peterbarker requested that we make these all const.

Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Viewpro.cpp]: All packets made const.

File: libraries/AP_Mount/AP_Mount_Viewpro.cpp (hunk 1 above)
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Viewpro.cpp:704]: I've retested with negative lat, lon and alt and the values appear correctly on the OSD so we know the gimbal is consuming them correctly even when negative.

File: libraries/AP_Mount/AP_Mount_Viewpro.cpp (hunk 2 above)
Comment (rmackay9) [reply libraries/AP_Mount/AP_Mount_Viewpro.cpp:706]: Also tested successfully with negative heights and they appeared correctly in the OSD.

**Comment by rmackay9** [comment]
Merged after clearing up issues raised on the EU dev call, thanks!
